OsmAnd/src/net/osmand/plus/mapcontextmenu/builders/FavouritePointMenuBuilder.java

```diff
@@ -76,11 +69,25 @@ public void buildInternal(View view) {
 		if (fav != null && fav.getTimestamp() != 0) {
 			buildDateRow(view, fav.getTimestamp());
 		}
-		if (originObject != null && originObject instanceof Amenity) {
-			AmenityMenuBuilder builder = new AmenityMenuBuilder(mapActivity, (Amenity) originObject);
-			builder.setLatLon(getLatLon());
-			builder.setLight(light);
-			builder.buildInternal(view);
+		if (!Algorithms.isEmpty(fav.getComment())) {
+			View rowc = buildRow(view, R.drawable.ic_action_note_dark, null, fav.getComment(), 0,
+					false, null, true, 10, false, null, false);
+			rowc.setOnClickListener(v -> POIMapLayer.showPlainDescriptionDialog(rowc.getContext(),
+					app, fav.getComment(), rowc.getResources().getString(R.string.poi_dialog_comment)));
+		}
+		if (originObject != null) {
+			Map<String, String> additionalInfo = null;
```

Comment:
1. Where is additional info from FavoritePoint itself?
2. How originObject could be Map or TransportStop if we only searched Amenity in acquireOriginObject?